### ISSUE TITLE
Exception generated: data must be a byte string - trying to add TV shows

### ISSUE THREAD

**Issue by whipcracker** [pymedusa/Medusa#1443]
### Before submitting your issue:

Enable debug logging in Medusa settings, reproduce the error (be sure to disable after the bug is fixed)

Branch/Commit: latest develop branch
OS: Linux-3.16.36-feral-x86_64-with-debian-8.6
What you did: Tried to add a show called "Humans" and tried "Narcos"
What happened: "search timed out, try again or try another indexer"
What you expected: For the show to appear and allow me to add to Medusa
Logs:
```
2016-11-02 02:38:18 ERROR    Thread-28 :: [9507206] Exception generated: data must be a byte string
Traceback (most recent call last):
  File "/media/sdq1/home/**********/.sickrage/medusa/server/web/core/base.py", line 271, in async_call
    result = function(**kwargs)
  File "/media/sdq1/home/**********/.sickrage/medusa/server/web/home/add_shows.py", line 80, in searchIndexersForShowName
    t = app.indexerApi(indexer).indexer(**l_indexer_api_parms)
  File "/media/sdq1/home/**********/.sickrage/medusa/indexers/indexer_api.py", line 34, in indexer
    return indexerConfig[self.indexer_id]['module'](*args, **kwargs)
  File "/media/sdq1/home/**********/.sickrage/medusa/indexers/tvdbv2/tvdbv2_api.py", line 66, in __init__
    access_token = auth_api.login_post(authentication_string)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/apis/authentication_api.py", line 124, in login_post
    callback=params.get('callback'))
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/api_client.py", line 324, in call_api
    response_type, auth_settings, callback)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/api_client.py", line 150, in __call_api
    post_params=post_params, body=body)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/api_client.py", line 360, in request
    body=body)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/rest.py", line 208, in POST
    body=body)
  File "/media/sdq1/home/**********/.sickrage/lib/tvdbapiv2/rest.py", line 143, in request
    headers=headers)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/request.py", line 73, in request
    **urlopen_kw)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/request.py", line 151, in request_encode_body
    return self.urlopen(method, url, **extra_kw)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/poolmanager.py", line 165, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connectionpool.py", line 578, in urlopen
    chunked=chunked)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/connectionpool.py", line 362, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/lib/python2.7/httplib.py", line 1039, in request
    self._send_request(method, url, body, headers)
  File "/usr/lib/python2.7/httplib.py", line 1073, in _send_request
    self.endheaders(body)
  File "/usr/lib/python2.7/httplib.py", line 1035, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python2.7/httplib.py", line 879, in _send_output
    self.send(msg)
  File "/usr/lib/python2.7/httplib.py", line 855, in send
    self.sock.sendall(data)
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/contrib/pyopenssl.py", line 253, in sendall
    sent = self._send_until_done(data[total_sent:total_sent + SSL_WRITE_BLOCKSIZE])
  File "/media/sdq1/home/**********/.sickrage/lib/requests/packages/urllib3/contrib/pyopenssl.py", line 242, in _send_until_done
    return self.connection.send(data)
  File "/usr/lib/python2.7/dist-packages/OpenSSL/SSL.py", line 947, in send
    raise TypeError("data must be a byte string")
TypeError: data must be a byte string
```

**Comment by ratoaq2** [comment]
@whipcracker 
Can you try this branch: tvdbv2-no-unicode ?

```
git fetch origin
git checkout tvdbv2-no-unicode
git clean -d -f lib medusa
git reset --hard origin/tvdbv2-no-unicode
```

**Comment by whipcracker** [comment]
Yeah mate no dramas, give me half an hour or so... Just out and about at the moment.

**Comment by whipcracker** [comment]
Just moved onto the branch... hasn't fixed it, but showing:

"No results found, try a different search." (tried adding "Humans" and Narcos"... same issue with both)

Thanks for the ongoing help!
````
2016-11-02 07:37:10 INFO     Thread-16 :: [f5afe1d] Error searching for show: ('Show search failed in getting a result with error: %r', TypeError('data must be a byte string',))
2016-11-02 07:37:05 INFO     Thread-27 :: [f5afe1d] Error searching for show: ('Show search failed in getting a result with error: %r', TypeError('data must be a byte string',))
2016-11-02 07:36:47 INFO     Thread-33 :: [f5afe1d] Error searching for show: ('Show search failed in getting a result with error: %r', TypeError('data must be a byte string',))
````

**Comment by whipcracker** [comment]
UPDATE: Can now edit existing TV Shows though! Just added a scene exception for the TV show "Timeless (2016)" with an exception just called "Timeless" and edited fine :D

That's excellent :D

**Comment by ratoaq2** [comment]
If you're still on the same branch, can you update and try again?

```
git pull
```

**Comment by whipcracker** [comment]
Hey @ratoaq2 

Just shut down, did a git pull and updated to:

 git log -1
commit f5afe1daa3bc8c1da3778859d1d938835d68524d
Author: Rato <rato.aq2@gmail.com>
Date:   Wed Nov 2 07:33:53 2016 +0100

    Removing unicode literals from tvdbv2 api.

I can now search for TV shows :D "Humans" is now showing a list of shows now, which is excellent. Tried "Narcos" and that worked as well :)

**Comment by whipcracker** [comment]
Whilst I won't update through the GUI, it does say:

"There is a newer version available (you're 13 commits behind and 1 commit ahead)— Update Now"

Interesting.

**Comment by fernandog** [comment]
@whipcracker 

git reset --hard origin/tvdbv2-no-unicode
git pull
git clean -df medusa
git clean -df lib
start medusa

**Comment by ratoaq2** [comment]
@whipcracker 
So, you can edit shows, you can add shows. Do you notice any other issue related to this? Should I consider this one fixed?

**Comment by whipcracker** [comment]
@ratoaq2 
I'll just do another check and see if there's anything weird in the logs for you and will report back in a few minutes. I'm just going to do what @fernandog has suggested above and test from there.

Thanks

**Comment by whipcracker** [comment]
Can now confirm that this is fixed (yeeeewwww!)

Successfully added "Humans" and was able to do manual searches, which worked... no errors or warnings in the logs either :) Thanks for all your help @ratoaq2 and @fernandog once again :D

Will prob need the process to move back to Develop and then over to Master once these commits have been merged of course :)

**Comment by fernandog** [comment]
@whipcracker can you do a quick test?

search/add shows that has special chars in the name

See if is added ok

**Comment by whipcracker** [comment]
Searched  é, came back with multiple results, so looks like special characters work :)

**Comment by whipcracker** [comment]
As per @fernandog 's recommendations, I have added "ワカコ酒 (Anime)" into Medusa successfully. I've then removed the show successfully as well :)

**Comment by ratoaq2** [comment]
Great.

Fixed by #1448
